packages/itinerary-body/src/TransitLegBody/index.tsx

```diff
@@ -321,16 +321,12 @@ class TransitLegBody extends Component<Props, State> {
 
                   {showViewTripButton && (
                     <ViewTripButton
-                      tripId={leg.tripId}
-                      fromIndex={
-                        leg.from.stopIndex ||
-                        leg?.trip?.departureStoptime?.stopPosition
-                      }
+                      fromIndex={leg.from?.stopIndex}
+                      fromStopId={leg?.from?.stopId}
```

Comment:
I think I know what is going on. The "from" and "to" stop ids come in OTP1 and OTP2 flavors. The code change in this file only cover the OTP1 response. The change you made to the mocks cover OTP2 response, which is not covered by this code, so please handle that. You need to add `leg.trip?.fromStopId` then fall back on `leg.from?.stopId`.

Reply:
The problem is the actual re-writing of the response is handled by OTP-RR. I can add a `stopId` to the mock, but this wouldn't represent actual OTP2 responses. It relies on otp-rr rewriting the responses

Reply:
In that case revert the mocks I made you change.

Reply:
Done!